File: backend/app/schemas/alert.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AlertRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    team_id: int
    pokemon_id: int
    move_id: int | None
    pokemon_change_log_id: int | None
    move_change_log_id: int | None
    message: str
    created_at: datetime

File: frontend/src/types/alert.ts
export interface Alert {
  id: number;
  team_id: number;
  pokemon_id: number;
  move_id: number | null;
  pokemon_change_log_id: number | null;
  move_change_log_id: number | null;
  message: string;
  created_at: string;
}
